client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
```

Comment:
The whole point of the message is about the domains we don't support though. A straightforward, non-optimized way could be to count the dots and make use of the `isSubdomain` function;
```
if (dots.length == 1) {
  // TLD
} else if ( isSubdomain( domain ) ) {
  // TLD (also subdomain)
} else {
  // SLD (grab last 2 dots)
}
```